src/phoenix/server/api/helpers/evaluators.py

```diff
@@ -0,0 +1,160 @@
+    except ValidationError as error:
+        raise ValueError(
+            _parse_pydantic_validation_error(
+                function_name=prompt_tool_function_definition.name,
+                validation_error=error,
+            )
+        )
+    function_property_name = next(iter(function_parameters["properties"].keys()))
+    if function_property_name != evaluator.annotation_name:
+        raise ValueError(
+            _LLMEvaluatorPromptErrorMessage.EVALUATOR_ANNOTATION_NAME_MUST_MATCH_FUNCTION_PROPERTY_NAME.value
+        )
+    function_property_choices = function_parameters["properties"][function_property_name]["enum"]
+    evaluator_choices = [value.label for value in evaluator.output_config.values]
+    if set(function_property_choices) != set(evaluator_choices):
+        raise ValueError(
+            _LLMEvaluatorPromptErrorMessage.EVALUATOR_CHOICES_MUST_MATCH_FUNCTION_PROPERTY_ENUM.value
+        )
+
+
+class _EvaluatorPromptToolFunctionParametersProperty(BaseModel):
+    type: Literal["string"]
+    enum: list[str] = Field(
+        ...,
+        min_length=2,
+    )
+
+
+class _EvaluatorPromptToolFunctionParameters(BaseModel):
+    type: Literal["object"]
+    properties: dict[str, _EvaluatorPromptToolFunctionParametersProperty] = Field(
+        ...,
+        min_length=1,
+        max_length=1,  # this constraint can be lifted to add support for multi-criteria evaluators
+    )
+    required: list[str]
+
+    @field_validator("required")
+    @classmethod
+    def check_required_values_are_unique(cls, values: list[str]) -> list[str]:
+        if len(values) != len(set(values)):
+            raise ValueError(_LLMEvaluatorPromptErrorMessage.REQUIRED_VALUES_MUST_BE_UNIQUE.value)
+        return values
+
+    @model_validator(mode="after")
+    def check_all_properties_are_required(self) -> Self:
+        defined_properties = set(self.properties)
+        required_properties = set(self.required)
+        if defined_properties - required_properties:
+            raise ValueError(
+                _LLMEvaluatorPromptErrorMessage.ALL_DEFINED_PROPERTIES_MUST_BE_REQUIRED.value
+            )
+        if required_properties - defined_properties:
+            raise ValueError(
+                _LLMEvaluatorPromptErrorMessage.ALL_REQUIRED_PROPERTIES_SHOULD_BE_DEFINED.value
+            )
+        return self
+
+
+def _parse_pydantic_validation_error(
+    function_name: str,
+    validation_error: ValidationError,
+) -> str:
+    error_messages = [f"'{function_name}' function has errors."]
+    for error_details in validation_error.errors():
+        error_message = ""
+        if path := ".".join(map(str, error_details["loc"])):
+            error_message = f"At '{path}': "
+        error_details_message = error_details["msg"]
+        error_message += f"{error_details_message}."
+        error_messages.append(error_message)
+    return " ".join(error_messages)
+
+
+class _LLMEvaluatorPromptErrorMessage(Enum):
```

Comment:
nit. doesn't need to be an Enum
```suggestion
class _LLMEvaluatorPromptErrorMessage:
```